File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/LoanRepaymentScheduleAdapter.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.mifosxdroid.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import com.mifos.mifosxdroid.R;
import com.mifos.mifosxdroid.databinding.RowLoanRepaymentScheduleBinding;
import com.mifos.objects.accounts.loan.Period;
import com.mifos.utils.DateHelper;

import java.util.List;



/**
 * Created by ishankhanna on 19/06/14.
 */
public class LoanRepaymentScheduleAdapter extends BaseAdapter {

    List<Period> periodList;
    LayoutInflater layoutInflater;
    Context context;

    public LoanRepaymentScheduleAdapter(Context context, List<Period> periodList) {

        layoutInflater = LayoutInflater.from(context);
        this.periodList = periodList;
        this.context = context;

    }

    @Override
    public int getCount() {
        return periodList.size();
    }

    @Override
    public Object getItem(int i) {
        return null;
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {

        ReusableViewHolder reusableViewHolder;
        RowLoanRepaymentScheduleBinding binding;

        if (view == null) {
            binding = RowLoanRepaymentScheduleBinding
                    .inflate(layoutInflater, viewGroup, false);
            reusableViewHolder = new ReusableViewHolder(binding);
            reusableViewHolder.view = binding.getRoot();
            reusableViewHolder.view.setTag(reusableViewHolder);
        } else {
            reusableViewHolder = (ReusableViewHolder) view.getTag();
            reusableViewHolder.view = view;
        }

        reusableViewHolder.tv_repaymentDueDate.setText(DateHelper.getDateAsString(periodList.get
                (i).getDueDate()));
        reusableViewHolder.tv_repaymentAmountDue.setText(String.valueOf(periodList.get(i)
                .getTotalDueForPeriod()));
        reusableViewHolder.tv_repaymentAmountPaid.setText(String.valueOf(periodList.get(i)
                .getTotalPaidForPeriod()));

        if (periodList.get(i).getComplete()) {
            reusableViewHolder.view_status_indicator.setBackgroundColor(
                    ContextCompat.getColor(context, R.color.light_green));

        } else if (!periodList.get(i).getComplete()
                && (periodList.get(i).getTotalOverdue() != null && periodList.get(i)
                .getTotalOverdue() > 0)) {
            reusableViewHolder.view_status_indicator.setBackgroundColor(
                    ContextCompat.getColor(context, R.color.light_red));
        } else {
            reusableViewHolder.view_status_indicator.setBackgroundColor(
                    ContextCompat.getColor(context, R.color.light_blue));

        }


        return reusableViewHolder.view;
    }

    public static class ReusableViewHolder {


        private View view;
        View view_status_indicator;
        TextView tv_repaymentDueDate;
        TextView tv_repaymentAmountDue;
        TextView tv_repaymentAmountPaid;

        public ReusableViewHolder(RowLoanRepaymentScheduleBinding binding) {
            view_status_indicator = binding.viewStatusIndicator;
            tv_repaymentDueDate = binding.tvRepaymentDueDate;
            tv_repaymentAmountDue = binding.tvRepaymentAmountDue;
            tv_repaymentAmountPaid = binding.tvRepaymentAmountPaid;
        }

    }

}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/LoanRepaymentScheduleAdapter.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.mifosxdroid.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import androidx.core.content.ContextCompat
import com.mifos.core.objects.accounts.loan.Period
import com.mifos.mifosxdroid.R
import com.mifos.mifosxdroid.databinding.RowLoanRepaymentScheduleBinding
import com.mifos.utils.DateHelper.getDateAsString

/**
 * Created by ishankhanna on 19/06/14.
 */
class LoanRepaymentScheduleAdapter(
    private val context: Context,
    private val periodList: List<Period>
) : BaseAdapter() {

    override fun getCount(): Int {
        return periodList.size
    }

    override fun getItem(i: Int): Any? {
        return null
    }

    override fun getItemId(i: Int): Long {
        return 0
    }

    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
        val viewHolder: ReusableViewHolder
        val binding: RowLoanRepaymentScheduleBinding

        if (view == null) {
            binding = RowLoanRepaymentScheduleBinding.inflate(
                LayoutInflater.from(context),
                viewGroup,
                false
            )
            viewHolder = ReusableViewHolder(binding)
            binding.root.tag = viewHolder
        } else {
            binding = RowLoanRepaymentScheduleBinding.bind(view)
            viewHolder = view.tag as ReusableViewHolder
        }

        val period = periodList[i]
        viewHolder.binding.tvRepaymentDueDate.text = period.dueDate?.let { getDateAsString(it) }
        viewHolder.binding.tvRepaymentAmountDue.text = period.totalDueForPeriod.toString()
        viewHolder.binding.tvRepaymentAmountPaid.text = period.totalPaidForPeriod.toString()

        when {
            period.complete != null && period.complete!! -> {
                viewHolder.binding.viewStatusIndicator.setBackgroundColor(
                    ContextCompat.getColor(
                        context,
                        R.color.light_green
                    )
                )
            }

            period.totalOverdue != null && period.totalOverdue!! > 0 -> {
                viewHolder.binding.viewStatusIndicator.setBackgroundColor(
                    ContextCompat.getColor(
                        context,
                        R.color.light_red
                    )
                )
            }

            else -> {
                viewHolder.binding.viewStatusIndicator.setBackgroundColor(
                    ContextCompat.getColor(
                        context,
                        R.color.light_blue
                    )
                )
            }
        }

        return binding.root
    }

    class ReusableViewHolder(val binding: RowLoanRepaymentScheduleBinding)
}